rosidl_generator_py/resource/_msg_support.c.em

```diff
@@ -248,156 +248,187 @@ nested_type = '__'.join(type_.namespaced_name())
     }
 @[    end if]@
 @[  elif isinstance(member.type, AbstractNestedType)]@
-@[    if isinstance(member.type, Array) and isinstance(member.type.value_type, BasicType) and member.type.value_type.typename in SPECIAL_NESTED_BASIC_TYPES]@
-    // TODO(dirk-thomas) use a better way to check the type before casting
-    assert(field->ob_type != NULL);
-    assert(field->ob_type->tp_name != NULL);
-    assert(strcmp(field->ob_type->tp_name, "numpy.ndarray") == 0);
-    PyArrayObject * seq_field = (PyArrayObject *)field;
-    Py_INCREF(seq_field);
-    assert(PyArray_NDIM(seq_field) == 1);
-    assert(PyArray_TYPE(seq_field) == @(SPECIAL_NESTED_BASIC_TYPES[member.type.value_type.typename]['dtype'].replace('numpy.', 'NPY_').upper()));
-@[    else]@
-    PyObject * seq_field = PySequence_Fast(field, "expected a sequence in '@(member.name)'");
-    if (!seq_field) {
-      Py_DECREF(field);
-      return false;
-    }
-@[    end if]@
-@[    if isinstance(member.type, AbstractSequence)]@
-    Py_ssize_t size = PySequence_Size(field);
-    if (-1 == size) {
-      Py_DECREF(seq_field);
-      Py_DECREF(field);
-      return false;
-    }
-@[      if isinstance(member.type.value_type, AbstractString)]@
-    if (!rosidl_runtime_c__String__Sequence__init(&(ros_message->@(member.name)), size)) {
-      PyErr_SetString(PyExc_RuntimeError, "unable to create String__Sequence ros_message");
-      Py_DECREF(seq_field);
-      Py_DECREF(field);
-      return false;
-    }
-@[      elif isinstance(member.type.value_type, AbstractWString)]@
-    if (!rosidl_runtime_c__U16String__Sequence__init(&(ros_message->@(member.name)), size)) {
-      PyErr_SetString(PyExc_RuntimeError, "unable to create U16String__Sequence ros_message");
-      Py_DECREF(seq_field);
-      Py_DECREF(field);
-      return false;
-    }
-@[      else]@
-    if (!rosidl_runtime_c__@(member.type.value_type.typename)__Sequence__init(&(ros_message->@(member.name)), size)) {
-      PyErr_SetString(PyExc_RuntimeError, "unable to create @(member.type.value_type.typename)__Sequence ros_message");
-      Py_DECREF(seq_field);
-      Py_DECREF(field);
-      return false;
-    }
-@[      end if]@
-    @primitive_msg_type_to_c(member.type.value_type) * dest = ros_message->@(member.name).data;
-@[    else]@
-    Py_ssize_t size = @(member.type.size);
-    @primitive_msg_type_to_c(member.type.value_type) * dest = ros_message->@(member.name);
-@[    end if]@
-    for (Py_ssize_t i = 0; i < size; ++i) {
-@[    if not isinstance(member.type, Array) or not isinstance(member.type.value_type, BasicType) or member.type.value_type.typename not in SPECIAL_NESTED_BASIC_TYPES]@
-      PyObject * item = PySequence_Fast_GET_ITEM(seq_field, i);
-      if (!item) {
-        Py_DECREF(seq_field);
+@[    if isinstance(member.type, AbstractSequence) and isinstance(member.type.value_type, BasicType)]@
```

Comment:
Lines 251-280 is a new code. Lines 281-430 is the old code with additional level of indentation, used as backup if PyObject_CheckBuffer call fails.

Reply:
Just as a note for interested parties - github diff view has "Ignore whitespace changes" which makes the diff a lot easier to look at in this case